app/routers/sessions.py

```diff
@@ -78,30 +96,54 @@ async def create_session(session: Session):
     return JSONResponse(status_code=status.HTTP_201_CREATED, content=created_session)
 
 
-@router.patch("/{session_id}", response_model=SessionResponse)
-async def update_session(session_id: str, session: UpdateSession):
-    session = jsonable_encoder(session)
-
-    if "has_quiz_ended" not in session:
-        raise HTTPException(
-            status_code=status.HTTP_400_BAD_REQUEST,
-            detail="No value provided for 'has_quiz_ended'",
-        )
-
-    if (client.quiz.sessions.find_one({"_id": session_id})) is None:
+@router.patch("/{session_id}", response_model=UpdateSessionResponse)
+async def update_session(session_id: str, session_updates: UpdateSession):
+    """
+    session is updated whenever start/resume button is clicked
+    (or) end button is clicked for the first time
+    """
+    session_updates = jsonable_encoder(session_updates)
+    session = client.quiz.sessions.find_one({"_id": session_id})
+    if session is None:
         raise HTTPException(
             status_code=status.HTTP_404_NOT_FOUND,
             detail=f"session {session_id} not found",
         )
 
-    # update the document in the sessions collection
-    client.quiz.sessions.update_one({"_id": session_id}, {"$set": session})
+    current_time = datetime.utcnow()
+    time_elapsed = 0
+
+    if session_updates["has_quiz_started_first_time"]:
+        # start/resume button clicked first time for a quiz
+        session["has_quiz_started"] = True
+        # time elapsed is zero
+    else:
+        # update time remaining based on current time (resume/end button clicked)
+        time_elapsed = (
+            current_time - datetime.fromisoformat(session["quiz_start_resume_time"])
+        ).seconds
+
+    if session_updates["has_quiz_ended_first_time"]:
+        # end test clicked first time
+        session["has_quiz_ended"] = True
+        session["quiz_end_time"] = current_time
+    else:
+        # quiz started or resumed
+        session["quiz_start_resume_time"] = current_time
```

Comment:
Should be separate values for `start` and `resume` (`resume` shoulnd be an array) so that we can track when the user started, stopped over multiple sessions.

Unless I'm missing something and that's already being tracked somehow

Reply:
so the `session["created"]` time will handle it. we could query for list of sessions of particular user/quiz pair - and look at array of times. however, the time captured in `created` pertains to when the quiz webpage has opened.

`start` and `resume` refer to when the user has "clicked" the start/resume button. ideally, they must be stored in an array, yes. but for current logic i didn't require. 

moreover, session start/resume times (as in, when the webpage has opened) -- can help for now.